apisix/utils/batch-processor.lua

```diff
@@ -181,7 +186,7 @@ function batch_processor:push(entry)
         return
     end
 
-    if not batch_metrics and prometheus.get_prometheus() and self.name
+    if is_http and not batch_metrics and prometheus.get_prometheus() and self.name
```

Comment:
Check if prometheus exists will be more straightforward?

Reply:
update